packages/tools/viewer-alpha/src/viewerElement.ts

```diff
@@ -170,20 +171,72 @@ export class HTML3DElement extends LitElement {
             pointer-events: none;
         }
 
-        .tool-bar {
+        .bar {
             position: absolute;
-            display: flex;
-            flex-direction: row;
-            border-radius: 12px;
-            border-color: var(--ui-foreground-color);
-            height: 48px;
             width: calc(100% - 24px);
             min-width: 150px;
             max-width: 1280px;
-            bottom: 12px;
             left: 50%;
             transform: translateX(-50%);
             background-color: var(--ui-background-color);
+        }
+
+        .loading-progress-outer {
+            height: 4px;
+            border-radius: 2px;
+            border: none;
+            outline: none;
+            top: 12px;
+            pointer-events: none;
+            transition: opacity 0.5s ease;
+        }
+
+        .loading-progress-outer-inactive {
+            opacity: 0;
+            /* Set the background color to the foreground color while in the inactive state so that the color seen is correct while fading out the opacity. */
+            background-color: var(--ui-foreground-color);
+        }
+
+        .loading-progress-inner {
+            width: 0;
+            height: 100%;
+            border-radius: inherit;
+            background-color: var(--ui-foreground-color);
+            transition: width 0.3s linear;
+        }
+
+        @keyframes indeterminate {
+            0% {
+                left: 0%;
+                width: 0%;
+            }
+            20% {
+                left: 0%;
+                width: 30%;
+            }
+            80% {
+                left: 70%;
+                width: 30%;
+            }

```

Comment:
i just wonder why it doesn't match the actual % of the keyframe?

Reply:
Math!